### PR TITLE
Details view improvements

### PR DESCRIPTION
**Pull request by gave92**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- For #4643

**Details of Changes**
Add details of changes here.
- Removed unused RectangleSelection_DataGrid, CustomDataGridStyle classes
- Added sort direction indicators to column headers
- Added size and date created columns
- Fixed an issue with details view icon column width

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

**Comment by yaichenbaum** [comment]
@gave92 It looks like the icon column is too wide now.

**Comment by gave92** [comment]
Should be same as before: 44px

image

**Comment by yaichenbaum** [comment]
@gave92 I think there is some extra padding already there because so it's not needed.

**Comment by gave92** [comment]
@yair100 ~could you check the new icon column width? I've changed it back to 24px, but looks kinda cramped.~ Nevermind I think I've messed up some paddings.
image

**Comment by yair100** [comment]
> @yair100 ~could you check the new icon column width? I've changed it back to 24px, but looks kinda cramped.~ Nevermind I think I've messed up some paddings.
> image

You can add a fixed width to the icon content so that it doesn't shift after the content is loaded.

**Comment by gave92** [comment]
I hate UIs, could you figure out how to make it stop moving when the folder is empty?

**Comment by yaichenbaum** [comment]
> I hate UIs, could you figure out how to make it stop moving when the folder is empty?

I can try. By the way, can we allow sorting by sync status?

**Comment by gave92** [comment]
Probably but it was never implemented.. next PR :)

**Comment by yaichenbaum** [comment]
> Probably but it was never implemented.. next PR :) Disabled button in sync status looks bad, but if we're going to allow sorting by status it's not an issue.

Sounds good to me, File Explorer allows sorting by status so we might as well.

**Comment by yaichenbaum** [comment]
@gave92 Can you unload the sort icon when it's not being used? It's causing the text to be trimmed when it shouldn't be.

**Comment by yaichenbaum** [comment]
@gave92 I found the cause of the header shifting. It shifts because the header width depends on the size of all the column combined. When the user changes the directory to a directory with a different number of columns, the header needs to recalculate the correct width. I don't have any ideas on how to solve this right now.

**Comment by gave92** [comment]
Mmm not sure that's the same issue: what I see is that all columns shift to the left 12px if the folder is empty. Anyway can be fixed next.

File: Files/Views/LayoutModes/DetailsLayoutBrowser.xaml [hunk 1]
```diff
@@ -319,33 +307,30 @@
                                 <ColumnDefinition x:Name="Column8" Width="0" />
                             </Grid.ColumnDefinitions>
 
-                            <Grid Grid.Column="0" Width="44" />
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/DetailsLayoutBrowser.xaml:322]: Can you revert this change?

Comment (gave92) [reply Files/Views/LayoutModes/DetailsLayoutBrowser.xaml:322]: Nope if I do the columns headers and contents are not aligned

Comment (yaichenbaum) [reply Files/Views/LayoutModes/DetailsLayoutBrowser.xaml:322]: I'll take a look again. It seems like some folders have a different width for that column.

Comment (gave92) [reply Files/Views/LayoutModes/DetailsLayoutBrowser.xaml:322]: Maybe because it has been saved in preferences for some reason? You coud try disabling "Enable individual preferences for individual folders"

Comment (yaichenbaum) [reply Files/Views/LayoutModes/DetailsLayoutBrowser.xaml:322]: That seems to cause other issues.
image

Comment (yaichenbaum) [reply Files/Views/LayoutModes/DetailsLayoutBrowser.xaml:322]: > Maybe because it has been saved in preferences for those folders? You could try disabling "Enable individual preferences for individual folders" and see if they are all the same width.

Good catch, I'll give that a try.

Comment (yaichenbaum) [reply Files/Views/LayoutModes/DetailsLayoutBrowser.xaml:322]: Can we prevent the user setting from being changed for the icon column?

**Review by yair100 — APPROVED** [review]
Great work! LGTM
